src/nvim/api/private/helpers.c

```diff
@@ -404,9 +404,18 @@ void set_option_to(void *to, int type, String name, Object value, Error *err)
       return;
     }
 
-    set_option_value_for(name.data, 0, value.data.string.data,
-            opt_flags, type, to, err);
+    stringval = (char *)value.data.string.data;
   }
+
+  const scid_T save_current_SID = current_SID;
+  current_SID = channel_id == LUA_INTERNAL_CALL ? SID_LUA : SID_API_CLIENT;
+  current_channel_id = channel_id;
```

Comment:
Is it too much work to pass the channel id through the function calls, instead of a global?

Reply:
No, I was just trying to avoid making lots of changes. If you're alright with making changes to all of the following functions and their callers, then I can do that instead of adding a global:
```
set_option_to
set_option_value_for
set_option_value_err
set_option_value
set_string_option
did_set_string_option
set_option_scriptID_idx
```

Reply:
@b-r-o-c-k  I retract. 

A future improvement would be to organize these "context" globals in a single struct which can be save/restored with a single assignment.

Could start with just a couple and add more as we find more cases.